### PR TITLE
DOC-6244: Added and updated Corda 4.12 release notes

### PR DESCRIPTION
**Pull request by AnnaK-R3**
ENT preview: https://neil.preview.docs.r3.com/en/platform/corda/4.12/enterprise/release-notes-enterprise.html
COMMUNITY preview: https://neil.preview.docs.r3.com/en/platform/corda/4.12/community/release-notes.html

### CODE REVIEW

**Review by chriscochrane — APPROVED** [review]

File: content/en/platform/corda/4.12/community/release-notes.md [hunk 1]
```diff
@@ -18,233 +18,215 @@ tags:
 
 ---
 
-# Corda Community Edition 4.11 release notes
+# Corda Open Source Edition 4.12 release notes
 
-## Corda Community Edition 4.11.1 release notes
+The Corda Open Source Edition 4.12 release introduces upgrades to the Java and Kotlin versions, along with associated upgrade support. Apart from the features supporting the Java and Kotlin upgrade, no other major new features have been introduced. In this release Java has been upgraded to Java 17 from Java 8 and Kotlin has been upgraded to Kotlin 1.9.20 from 1.2.71.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.12/community/release-notes.md:23]:
```suggestion
The Corda Open Source Edition 4.12 release introduces upgrades to the Java and Kotlin versions, along with associated upgrade support. Apart from the features supporting the Java and Kotlin upgrade, no other major new features have been introduced. In this release, Java has been upgraded to Java 17 from Java 8 and Kotlin has been upgraded to Kotlin 1.9.20 from 1.2.71.
```

File: content/en/platform/corda/4.12/community/release-notes.md [hunk 2]
```diff
@@ -18,233 +18,215 @@ tags:
...
-## New features and enhancements
+## New features, enhancements and restrictions
+
+### Java and Kotlin upgrade
+
+Corda 4.12 requires Java 17 and Kotlin 1.9.20. This means that you must recompile any legacy CorDapps written for 4.11 or earlier to work with Java 17 and Kotlin 1.9 to be compatible with Corda 4.12. These upgrades enhance the supportability and security of Corda.
+
+### Java 17 compatible releases of Corda SDKs
+
+A number of SDKs and libraries are provided with the base Corda package. These are their Java 17 and Kotlin 1.9 versions:
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.12/community/release-notes.md:49]:
```suggestion
A number of SDKs and libraries are provided with the base Corda package. These are the Corda internal versions compatible with Java 17:
```

Not sure the original was correct, maybe this is?

File: content/en/platform/corda/4.12/community/release-notes.md [hunk 3]
```diff
@@ -18,233 +18,215 @@ tags:
...
+Explicit contract upgrade is not supported in Corda 4.12.
+
+### `toLedgerTransaction.verify` does not work for legacy transactions
+
+You must review your CorDapps and check for any making the following calls:
+* `SignedTransaction.toLedgerTransaction().verify()`
+* `WireTransaction.toLedgerTransaction().verify()`
+* `TransactionBuilder.toLedgerTransaction().verify()`
+
+CorDapps that make the above calls, will not work for legacy transactions. To have those CorDapps work, change them to `SignedTransaction.verify()`.
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.12/community/release-notes.md:82]:
```suggestion
CorDapps that make the above calls will not work for legacy transactions. To make those CorDapps compatible, change them to `SignedTransaction.verify()`.
```

File: content/en/platform/corda/4.12/community/release-notes.md [hunk 4]
```diff
@@ -18,233 +18,215 @@ tags:
...
+
+### Samples Kotlin and Java support
+
+The following two public repositories provide various CorDapp samples (branch: release/4.12):
+* [Samples Kotlin repository](https://github.com/corda/samples-kotlin/tree/release/4.12)
+* [Samples Java repository](https://github.com/corda/samples-java/tree/release/4.12)
+
+Most samples have been converted over to Java 17, Kotlin 1.9.20, and Gradle 7.6.4.
+
+The samples have been written to work with Corda Open Source. To convert a sample to work with Corda Enterprise at a minimum you need to point to a repository where your enterprise artifacts are installed and the artifact group name for ENT (`com.r3`) must be different from OS `(net.corda`). For example, switch `net.corda:corda-node-driver:4.12` (Corda OS) to `com.r3.corda:corda-node-driver:4.12` (Corda ENT).
```
Comment (paulmoloneyr3) [inline content/en/platform/corda/4.12/community/release-notes.md:96]:
```suggestion
The samples have been written to work with Corda Open Source. To convert a sample to work with Corda Enterprise, then at a minimum you need to point to a repository where your enterprise artifacts are installed. Also, the artifact group name for ENT (`com.r3`) must be different from OS `(net.corda`). For example, switch `net.corda:corda-node-driver:4.12` (Corda OS) to `com.r3.corda:corda-node-driver:4.12` (Corda ENT).
```

**Review by paulmoloneyr3 — CHANGES_REQUESTED** [review]

File: content/en/platform/corda/4.12/community/release-notes.md (hunk 2 above)
Comment (AnnaK-R3) [reply content/en/platform/corda/4.12/community/release-notes.md:49]: Yes, that might not be very clear. Rephrased.

**Review by paulmoloneyr3 — APPROVED** [review]

**Pull request merged**